sys/include/net/gnrc/netif/flags.h

```diff
@@ -23,15 +23,20 @@ extern "C" {
 #endif
 
 /**
- * @brief   Auto-address configuration modes
- * @anchor  net_gnrc_netif_aac
+ * @brief   Auto-address configuration mode flags
+ * @anchor  net_gnrc_netif_aac_flags
+ * @{
+ */
+/**
+ * @brief   Use some automatic bootstrapping (e.g. SLAAC with IPv6)
  */
-enum {
-    GNRC_NETIF_AAC_NONE = 0,    /**< no configuration */
-    GNRC_NETIF_AAC_AUTO,        /**< Use some automatic bootstrapping (e.g. SLAAC with IPv6) */
-    GNRC_NETIF_AAC_DHCP,        /**< Use DHCP(v6) */
-    /* extend if needed */
-};
+#define GNRC_NETIF_AAC_FLAGS_AUTO                  (0x00000001U)
+
+/**
+ * @brief   Use DHCP(v6)
+ */
+#define GNRC_NETIF_AAC_FLAGS_DHCP                  (0x00000002U)
+/** @} */
```

Comment:
The rename shouldn't be necessary. Also you can still use the `enum` to define the flags.

```suggestion
enum {
    GNRC_NETIF_AAC_NONE = 0,    /**< no configuration */
    GNRC_NETIF_AAC_AUTO = 0x01, /**< Use some automatic bootstrapping (e.g. SLAAC with IPv6) */
    GNRC_NETIF_AAC_DHCP = 0x02, /**< Use DHCP(v6) */
    /* extend if needed */
};
/** @} */
```

Lastly, `aac_mode` is a `uint8_t`, so way to many zeros in front ;-).

Reply:
Oh, I am sorry, there have been a couple of fails that have been incorporated into this commit 🤦‍♂️ Making a separate PR out of this change sounds very good, I will open one soon (with the fixed changes) :)

Reply:
I've opened #16596 now and would remove 3e91f680728cf67a673f559ddcd25667ef35b2fd from this PR :)